argilla/tests/unit/test_settings/test_settings_from_features.py

```diff
@@ -0,0 +1,120 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import argilla as rg
+
+from argilla._exceptions._settings import SettingsError
+from argilla.settings._io._hub import _define_settings_from_features
+
+
+def test_define_settings_from_features_text():
+    features = {"text_column": {"_type": "Value", "dtype": "string"}}
+    settings = _define_settings_from_features(features, feature_mapping={})
+
+    assert len(settings.fields) == 1
+    assert isinstance(settings.fields[0], rg.TextField)
+    assert settings.fields[0].name == "text_column"
+    assert len(settings.questions) == 1
+    assert isinstance(settings.questions[0], rg.TextQuestion)
+    assert settings.questions[0].name == "text_column_question"
+    assert settings.mapping == {}
+
+
+def test_define_settings_from_features_image():
+    features = {"image_column": {"_type": "Image"}}
+    settings = _define_settings_from_features(features, feature_mapping={})
+
+    assert len(settings.fields) == 1
+    assert isinstance(settings.fields[0], rg.ImageField)
+    assert settings.fields[0].name == "image_column"
+    assert len(settings.questions) == 1
+    assert settings.questions[0].name == "comment"
+    assert settings.mapping == {}
+
+
+def test_define_settings_from_features_multiple():
+    features = {
+        "text_column": {"_type": "Value", "dtype": "string"},
+        "image_column": {"_type": "Image"},
+        "label_column": {"_type": "ClassLabel", "names": ["A", "B"]},
+    }
+    settings = _define_settings_from_features(features, feature_mapping={})
+
+    assert len(settings.fields) == 2
+    assert isinstance(settings.fields[0], rg.TextField)
+    assert settings.fields[0].name == "text_column"
+    assert isinstance(settings.fields[1], rg.ImageField)
+    assert settings.fields[1].name == "image_column"
+    assert len(settings.questions) == 2
+    assert isinstance(settings.questions[0], rg.TextQuestion)
+    assert settings.questions[0].name == "text_column_question"
+    assert isinstance(settings.questions[1], rg.LabelQuestion)
+    assert settings.questions[1].name == "label_column"
+
+
+def test_mapped_question():
+    features = {
+        "text_column": {"_type": "Value", "dtype": "string"},
+        "image_column": {"_type": "Image"},
+        "label_column": {"_type": "ClassLabel", "names": ["A", "B"]},
+    }
+    settings = _define_settings_from_features(features, feature_mapping={"text_column": "question"})
+
+    assert len(settings.fields) == 1
+    assert isinstance(settings.fields[0], rg.ImageField)
+    assert settings.fields[0].name == "image_column"
+    assert len(settings.questions) == 2
+    assert isinstance(settings.questions[0], rg.TextQuestion)
+    assert settings.questions[0].name == "text_column_question"
+    assert isinstance(settings.questions[1], rg.LabelQuestion)
+    assert settings.questions[1].name == "label_column"
```

Comment:
Should be this the expected mapping?
```suggestion
    assert settings.questions[1].name == "label_column"
    assert settings.mapping == {  "text_column":  "text_columnt_question" }
```